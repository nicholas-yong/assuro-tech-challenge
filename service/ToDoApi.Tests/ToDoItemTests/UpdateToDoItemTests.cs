using System;
using System.Threading.Tasks;
using HotChocolate.Execution;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using NUnit.Framework;
using Snapshooter.NUnit;
using ToDoApi.Data;

namespace ToDoApi.Tests.ToDoItemTests
{
    public class UpdateToDoItemTests: BaseTest
    {
        [Test]
        public async Task CanUpdateItem()
        {
            // arrange
            var dbContext = Services
                .GetRequiredService<IDbContextFactory<ApplicationDbContext>>()
                .CreateDbContext();
            var toDoItem = new ToDoItem {
                Content = "My new todo",
                CreatedDate = DateTimeOffset.UtcNow,
                Id = Guid.NewGuid(),
                Status = ToDoItemStatus.PENDING,
            };
            dbContext.ToDoItems.Add(toDoItem);
            await dbContext.SaveChangesAsync();

            // act
            IExecutionResult result = await Services.ExecuteRequestAsync(
                QueryRequestBuilder.New()
                    .SetQuery(@"
                        mutation($id: Uuid!)
                        {
                            updateToDoItem(input: {
                                id: $id
                                content: ""My updated todo""
                            }) {
                                toDoItem {
                                    content
                                    createdDate
                                    id
                                    status
                                }
                            }
                        }")
                    .SetVariableValue("id", toDoItem.Id)
                    .Create());
            
            // assert
            result.MatchSnapshot(match => match
                .IgnoreField<Guid>("Data.updateToDoItem.toDoItem.id")
                .IgnoreField<DateTimeOffset>("Data.updateToDoItem.toDoItem.createdDate")
            );
        }
    }
}
