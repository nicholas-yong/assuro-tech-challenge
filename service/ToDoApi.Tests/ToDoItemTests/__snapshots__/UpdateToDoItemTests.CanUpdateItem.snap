{
  "Label": null,
  "Path": null,
  "Data": {
    "updateToDoItem": {
      "toDoItem": {
        "content": "My updated todo",
        "createdDate": "2021-02-17T04:05:53.314Z",
        "id": "b9cd9ace42e24df4b8253615dca39c34",
        "status": "PENDING"
      }
    }
  },
  "Errors": null,
  "Extensions": null,
  "ContextData": null,
  "HasNext": null
}
